builder-base/Dockerfile

```diff
@@ -15,13 +15,37 @@
 ARG BASE_IMAGE
 FROM ${BASE_IMAGE}
 
-COPY golang-checksum /golang-checksum
-COPY buildkit-checksum /buildkit-checksum
-COPY rootlesskit-checksum /rootlesskit-checksum
-COPY bazel-checksum /bazel-checksum
-COPY bash-checksum /bash-checksum
+ENV GOPATH /go
+ENV PATH="/go/bin/:$PATH"
+
+COPY *-checksum /
 COPY install.sh /opt/install.sh
+
 RUN bash /opt/install.sh
 
-ENV GOPATH /go
+# TODO: break the above install.sh script up into what is needed
```

Comment:
Talking with @micahhausler  the original purpose of using the install.sh script was because the pod building the builder-base needed the same deps as builder-base itself.  This makes sense, however, at this point there is way more in the install.sh than is needed to build the builder-base so the original intent is not clear unless it is explained.  In addition to that, adding to the builder-base is painful because using the install script method does not allow docker to cache layers.

I propose doing the following for everything that is not required to build the builder-base to get the best of both of these worlds.  The install.sh script can still be used to bootstrap the pod and the builder-base, but everything else can be individual layers making rebuilds locally when making modifications less painful.